src/analysis/bounds.rs

```diff
@@ -50,8 +55,16 @@ impl Bounds {
             _ => String::new(),
         }
     }
-    pub fn add_parameter(&mut self, name: &str, type_str: &str, bound_type: BoundType) -> bool {
-        if self.used.iter().any(|ref n| n.0 == name)  { return false; }
+    pub fn add_parameter(&mut self, name: &str, type_str: &str, bound_type: BoundType,
+                         is_nullable: bool) -> bool {
+        if self.used.iter().any(|ref n| n.0 == name) { return false; }
+        if bound_type == BoundType::Into {
+            if is_nullable == false { return true; }
```

Comment:
If `is_nullable` used only for this check, may be better move it to `type_for` or add separate check function that must called on `BoundType::Into`